### PR TITLE
[spaceship] Webpush Id support added for spaceship.

### PR DESCRIPTION
**Pull request by ilhanadiyaman**
Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes:

- [x] Run `bundle exec rspec` from the subdirectory of each tool you modified. Alternatively, run `rake test_all` from the root directory.
- [x] Run `bundle exec rubocop -a` to ensure the code style is valid
- [x] Read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] We currently don't accept new actions, please publish a plugin instead, more information in [Plugins.md](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Plugins.md)

Before submitting a pull request, we appreciate if you create an issue first to discuss the change :+1:

# API

## List all WebsitePush IDs:
returns array of Spaceship::Portal::WebsitePush
```ruby
Spaceship::Portal::WebsitePush.all
```
## Find by Website Push ID:
```ruby
website_push = Spaceship::Portal::WebsitePush.find("web.com.example.two")
```
## Create a WebsitePush ID:
```ruby
Spaceship::Portal::WebsitePush.create!(bundle_id: 'web.com.fastlane.example', name: 'Fastlane Website Push')
```
## Delete a WebsitePush ID:
```ruby
website_push = Spaceship::Portal::WebsitePush.find("web.com.example.two")
website_push.delete!
```

### CODE REVIEW

**Comment by ohwutup** [comment]
Hey @ilhanadiyaman! Thanks for the PR! Would you mind just writing down a bit of a summary of these changes? Thanks!

**Comment by ohwutup** [comment]
Hey @ilhanadiyaman! Thanks for this PR! I am going to go ahead and close it for now because there has not been any activity on it in quite some time. Please feel free to provide more context around why this PR is helpful and we would be happy to reopen it and consider getting in merged in! Feel free to ping any of us on here when you have the time! Thanks again! 🚀 🐙

**Comment by RafaelPlantard** [comment]
I like it!

**Comment by ilhanadiyaman** [comment]
Hi @ohwutup! Sorry for my late response, I forgot to add my comments. This PR is actually adds web push notification identifier creating/listing/deleting operations to spaceship. Web push identifiers are another entities in the portal and it would be helpful for people who wants to automate their web push identifier generation process. In our [web push notification startup](https://frizbit.com/?utm_source=github), we need to automate this process. Because, most of our customers were having trouble with creating web push identifier via Apple Developer Portal. With the help of this PR, we solve this issue. This is why I thought there may be other people who were suffering from the same problem. Cheers! 🚀

File: spaceship/spec/portal/website_push_spec.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
+require 'spec_helper'
```
Comment (hjanuschka) [inline spaceship/spec/portal/website_push_spec.rb]: i dont think that is required

**Comment by RafaelPlantard** [comment]
@ilhanadiyaman is possible generate a web push identifier and also creating certificate (pem, p12) files?

**Review by hjanuschka — COMMENTED** [review]
@ilhanadiyaman looking good so far, i think its a good thing for spaceship.
left one comment. and could you please add samples on how to interact with the new api. like here: https://github.com/fastlane/fastlane/pull/8047

then i am going to test this 👍

**Review by hjanuschka — CHANGES_REQUESTED** [review]
see comment before - also can you please fix the merge conflicts.

**Comment by ilhanadiyaman** [comment]
@RafaelPlantard yes actually this is how we are using it on our startup right now. We are creating first the web push identifiers over spaceship then generating certificates by using pem(it already supports web push certificates no need to change.).

@hjanuschka I will take a look probably tonight or tomorrow.

**Comment by hjanuschka** [comment]
@ilhanadiyaman thats awesome so this is the missing piece to fully automate webpush things, maybe add a pem sample too ;)

**Comment by RafaelPlantard** [comment]
@ilhanadiyaman Can you show me how to generate a pem file with PEM? please, rsrs

**Comment by ilhanadiyaman** [comment]
@RafaelPlantard I will provide sample codes about our process. Hopefully, tomorrow..

**Comment by RafaelPlantard** [comment]
Thanks @ilhanadiyaman

**Comment by RafaelPlantard** [comment]
@ilhanadiyaman can you help me with the samples?

File: spaceship/spec/portal/website_push_spec.rb (hunk 1 above)
Comment (KrauseFx) [inline spaceship/spec/portal/website_push_spec.rb]: 👍

**Comment by ilhanadiyaman** [comment]
@hjanuschka ready for review. @RafaelPlantard I have checked my forked repository. But regretfully, in order to create certificates I just realized that I made changes on pem as well. Those changes are not well tested. This is why I couldn't added that feature to this pull request.

**Comment by RafaelPlantard** [comment]
Any release date to merge this PR @ohwutup?

**Review by hjanuschka — APPROVED** [review]
🚀 awesome! many thx for your contribution.

**Comment by hjanuschka** [comment]
@ilhanadiyaman am i right that only pem changes are missing, to complete the website push creation? have you found any time to finish it?

**Comment by ilhanadiyaman** [comment]
@hjanuschka Last month I couldn't find enough time to work on it. I will take a look at it to finalize pem testing. If I can't finish this week, I will pass it to you..

**Pull request merged**